src/api/endpoints/epochs.rs

```diff
@@ -0,0 +1,255 @@
+            ("https://docs.blockfrost.io/#tag/Cardano-Epochs/paths/~1epochs~1{number}~1blocks/get"),
+
+        /// Return the block minted for the epoch specified by stake pool.
+        epochs_blocks_by_pool(number: Integer, pool_id: &str) -> Vec<String> => "/epochs/{number}/blocks/{pool_id}";
+            ("https://docs.blockfrost.io/#tag/Cardano-Epochs/paths/~1epochs~1{number}~1blocks~1{pool_id}/get"),
+
+        /// Return the protocol parameters for the epoch specified
+        epochs_parameters(number: Integer) -> EpochParameters => "/epochs/{number}/parameters";
+            ("https://docs.blockfrost.io/#tag/Cardano-Epochs/paths/~1epochs~1{number}~1parameters/get"),
+    }
+}
+
+/// Created by [`epochs_latest`](BlockFrostApi::epochs_latest) method.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Epoch {
+    /// Epoch number.
+    pub epoch: Integer,
+    /// Unix time of the start of the epoch.
+    pub start_time: Integer,
+    /// Unix time of the end of the epoch.
+    pub end_time: Integer,
+    /// Unix time of the first block of the epoch.
+    pub first_block_time: Integer,
+    /// Unix time of the last block of the epoch.
+    pub last_block_time: Integer,
+    /// Number of blocks within the epoch.
+    pub block_count: Integer,
+    /// Number of transactions within the epoch.
+    pub tx_count: Integer,
+    /// Sum of all the transactions within the epoch in Lovelaces.
+    pub output: String,
+    /// Sum of all the fees within the epoch in Lovelaces.
+    pub fees: String,
+    /// Sum of all the active stakes within the epoch in Lovelaces.
+    pub active_stake: Option<String>,
+}
+
+/// Created by [`epochs_latest_parameters`](BlockFrostApi::epochs_latest_parameters) method.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct EpochParameters {
+    /// Epoch number
+    pub epoch: Integer,
+    /// The linear factor for the minimum fee calculation for a given epoch.
+    pub min_fee_a: Integer,
+    /// The constant factor for the minimum fee calculation.
+    pub min_fee_b: Integer,
+    /// Maximum block body size in Bytes.
+    pub max_block_size: Integer,
+    /// Maximum transaction size.
+    pub max_tx_size: Integer,
+    /// Maximum block header size.
+    pub max_block_header_size: Integer,
+    /// The amount of a key registration deposit in Lovelaces.
+    pub key_deposit: String,
+    /// The amount of a pool registration deposit in Lovelaces.
+    pub pool_deposit: String,
+    // Epoch bound on pool retirement.
+    pub e_max: Integer,
+    /// Desired number of pools.
+    pub n_opt: Integer,
+    /// Pool pledge influence.
+    pub a0: Float,
+    /// Monetary expansion.
+    pub rho: Float,
+    /// Treasury expansion.
+    pub tau: Float,
+    /// Percentage of blocks produced by federated nodes
+    pub decentralisation_param: Float,
+    // Seed for extra entropy. TODO: how to represent arbitrary objects?
+    // pub extra_entropy: Option<??>,
```

Comment:
Here you could use `Option<HashMap<String, JsonValue>>`, there is an example of it being used by us:

https://github.com/blockfrost/blockfrost-rust/blob/8ed95baf2720bcb59925c597920da13b3ec01e33/src/api/endpoints/assets.rs#L67

However, because this weird type is being used in multiple places now, it would be better to add it as a type alias to `types.rs` and document that it can be found in `EpochParameters` and `AssetDetails`.

```rust
pub type JsonObject = HashMap<String, JsonValue>;
```

Reply:
Can do!

Reply:
FYI, although this works, I gave a bad advice on using `HashMap` here.

A better suggestion I could have given was to use https://docs.serde.rs/serde_json/map/struct.Map.html instead.

As explained [here (variant docs)](https://docs.serde.rs/serde_json/value/enum.Value.html#variant.Object), `serde_json` gives a simpler interface for a `Map` that is backed-up with `BTreeMap` by default, but also enables changing it to [`IndexMap`](docs.rs/indexmap), which can be useful.